main.go

```diff
@@ -209,39 +492,104 @@ func runClient(host string) {
 		Transport: tr,
 	}
 
-	ctx, cancel := context.WithCancel(context.Background())
-	r := &netperfReader{doneCh: ctx.Done()}
-	r.buf = make([]byte, 1*humanize.MiByte)
-	rand.Read(r.buf)
+	var index string
+	for {
+		time.Sleep(1 * time.Second)
+		index = getWriterIndex(r, clnt)
+		if index == "" {
+			globalErr = errors.New("Waiting for index ...")
+			if ctx.Err() != nil {
+				return
+			}
+		} else {
+			break
+		}
+	}
 
 	var wg sync.WaitGroup
 	for i := 0; i < proc; i++ {
+		if ctx.Err() != nil {
+			break
+		}
```

Comment:
Is this needed? We aren't really blocking here, right? We are blocking on the `wg.Wait()` below.